thingsvision/core/extraction/helpers.py

```diff
@@ -1,8 +1,193 @@
+
+import torch
 import numpy as np
+import tensorflow as tf
+
+import thingsvision.custom_models as custom_models
+import thingsvision.custom_models.cornet as cornet
 
+from typing import Any, Dict, Callable
+from .base import BaseExtractor
+from .mixin import PyTorchMixin, TensorFlowMixin
+from .extractor import TorchvisionExtractor, TimmExtractor, KerasExtractor
+
+Tensor = torch.Tensor
 Array = np.ndarray
 AxisError = np.AxisError
 
+def create_custom_extractor(
+    model_name: str,
+    pretrained: bool,
+    device: str,
+    model_path: str = None,
+    model_parameters: Dict[str, str] = None
+) -> Any:
+    """Create a custom extractor from a pretrained model."""
+    if model_name.startswith("cornet"):
+        backend = "pt"
+        try:
+            model = getattr(cornet, f"cornet_{model_name[-1]}")
+        except AttributeError:
+            model = getattr(cornet, f"cornet_{model_name[-2:]}")
+        model = model(
+            pretrained=pretrained, map_location=torch.device(device)
+        )
+        model = model.module  # remove DataParallel
+        preprocess = None
+    elif hasattr(custom_models, model_name):
+        custom_model = getattr(custom_models, model_name)
+        custom_model = custom_model(device, model_parameters)
+        model, preprocess = custom_model.create_model()
+        backend = custom_model.get_backend()
+    else:
+        raise ValueError(
+            f"\nCould not find {model_name} among custom models.\nChoose a different model.\n"
+        )
+
+    backend_mixin = PyTorchMixin if backend == "pt" else TensorFlowMixin
+
+    class CustomExtractor(BaseExtractor, backend_mixin):
+        def __init__(self, *args, **kwargs) -> None:
+            super().__init__(*args, **kwargs)
+
+    #TODO: this should probably be defined in the custom model itself
+    if model_name.lower().startswith('clip'):  
```

Comment:
We should add another issue to make this a bit nicer (i.e. putting these model-specific functions into the corresponding custom_model files)

Reply:
Totally agree. This `if-statement` is a bit ugly. It does its job, but could definitely be made nicer. We should move it into the corresponding `custom_model`.

Reply:
Add an additional issue that addresses all source-/model-specific details. We want to move everything model-/source-specific to `custom_model` or the `source`-specific extractor class. Nothing that's source-/model-specific should be in `BaseExtractor` or `PyTorchMixin`/`TensorFlowMixin`. Since the above addresses all `clip` models, we could think about moving this to the `OpenCLIP` extractor.